libs/forecasting/src/lib/ForecastingDashboard/ForecastingDashboard.tsx

```diff
@@ -41,38 +38,65 @@ export class ForecastingDashboard extends React.Component<
     super(props);
 
     this.state = {
-      isTransformationCreatorVisible: false
+      isTransformationCreatorVisible: false,
+      whatIfEnabled: true
     };
   }
 
+  public componentDidMount(): void {
+    const timeSeries = this.context.errorCohorts.filter(
+      (cohort) => !isAllDataErrorCohort(cohort)
+    );
+    const dropdownOptions: IDropdownOption[] = timeSeries.map((cohort) => {
+      return {
+        key: cohort.cohort.getCohortID(),
+        text: cohort.cohort.name
+      };
+    });
+    // If there is only 1 time series, switch to the corresponding cohort.
+    if (timeSeries.length === 1) {
+      this.context.shiftErrorCohort(timeSeries[0]);
+    }
+    // If the dataset only has a time column and time series identifying columns
+    // there is nothing to transform with what-if.
+    const whatIfEnabled = featureColumnsExist(
+      this.context.dataset.feature_names,
+      this.context.dataset.feature_metadata
+    );
+
+    this.setState({ timeSeriesOptions: dropdownOptions, whatIfEnabled });
+  }
+
   public render(): React.ReactNode {
     const classNames = forecastingDashboardStyles();
 
     if (
       this.context === undefined ||
-      this.context.baseErrorCohort === undefined
+      this.context.baseErrorCohort === undefined ||
+      this.state.timeSeriesOptions === undefined
     ) {
-      return;
+      return <></>;
     }
 
     // "All data" cohort selected, so no particular time series selected yet.
-    // special case: only 1 time series in dataset, needs to be handled! TODO
-    const noCohortSelected = isAllDataErrorCohort(this.context.baseErrorCohort);
-
-    const dropdownOptions: IDropdownOption[] = this.context.errorCohorts
-      .filter((cohort) => !isAllDataErrorCohort(cohort))
-      .map((cohort) => {
-        return {
-          key: cohort.cohort.getCohortID(),
-          text: cohort.cohort.name
-        };
-      });
+    // Special case: if there's only 1 time series then it is selected by default.
+    const noCohortSelected =
+      isAllDataErrorCohort(this.context.baseErrorCohort) &&
+      this.state.timeSeriesOptions.length > 1;
 
     const cohortTransformations =
       this.state.transformations?.get(
         this.context.baseErrorCohort.cohort.getCohortID()
       ) ?? new Map<string, Transformation>();
 
+    const description =
+      (this.state.whatIfEnabled
```

Comment:
nit: a small helper function might make this description creation a little clearer

Reply:
Fair 🤣